src/components/tr-logs.tsx

```diff
@@ -0,0 +1,151 @@
+								})}
+							</div>
+							:
+							log.event === "artifactserver.getArtifactID" ?
+								<div>
+									{translations.page.rich("showSciperLogMessage", {
+										badge: (chunks) => <span className={`inline-flex items-center px-2.5 py-0.5 rounded-none text-xs font-medium ${getArtifactBadgeColor(metadata?.status as number)}`}>{chunks}</span>,
+										targetEmail: targetId,
+										sciper: targetResult,
+										errorMessage: (error?.errorMessage || "undefined") as string,
+										bold: (chunks) => <b className="font-bold">{chunks}</b>,
+									})}
+								</div>
+								:
+								<>{log.event}</>
+				}
+				{
+					metadata?.soapRequest && (
+						<div className="flex justify-between">
+							<Dialog>
+								<DialogTrigger asChild>
+									<button className="hover:cursor-pointer">
+										<CodeXml className="h-5 w-5" />
+									</button>
+								</DialogTrigger>
+								<DialogContent className="">
+									<DialogHeader>
+										<DialogTitle>
+											aa
+										</DialogTitle>
+									</DialogHeader>
+									<Tabs defaultValue="response" className="">
+										<TabsList>
+											<TabsTrigger value="request">Request</TabsTrigger>
+											<TabsTrigger value="response">Response</TabsTrigger>
+										</TabsList>
+										<TabsContent value="request">
+											<Card>
+												<CardContent className="grid p-3">
+													<CodeBlock code={metadata?.soapRequest as string} language="xml" theme={themes.jettwaveLight}>
+														<CodeBlock.Code className="bg-white p-6 rounded-xl shadow-lg text-xs overflow-x-auto w-full">
+															<div className="table-row">
+																<CodeBlock.LineNumber className="table-cell pr-4 text-sm text-gray-500 text-right select-none" />
+																<CodeBlock.LineContent className="table-cell">
+																	<CodeBlock.Token />
+																</CodeBlock.LineContent>
+															</div>
+														</CodeBlock.Code>
+													</CodeBlock>
+												</CardContent>
+											</Card>
+										</TabsContent>
+										<TabsContent value="response">
+											<Card>
+												<CardContent className="grid p-3">
+													<CodeBlock code={metadata?.soapResponse as string} language="xml" theme={themes.jettwaveLight}>
+														<CodeBlock.Code className="bg-white p-6 rounded-xl shadow-lg text-xs overflow-x-auto w-full">
+															<div className="table-row">
+																<CodeBlock.LineNumber className="table-cell pr-4 text-sm text-gray-500 text-right select-none" />
+																<CodeBlock.LineContent className="table-cell">
+																	<CodeBlock.Token />
+																</CodeBlock.LineContent>
+															</div>
+														</CodeBlock.Code>
+													</CodeBlock>
+												</CardContent>
+											</Card>
+										</TabsContent>
+									</Tabs>
+								</DialogContent>
+							</Dialog>
+						</div>
+					)
+				}
+			</td>
+		</tr>
+	);
+}
```

Comment:
It [always a good idea to have an empty line at the end](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file).